frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortletUtils.java

```diff
@@ -369,6 +371,27 @@ public static VerificationState getVerificationState(VulnerabilityDTO vul){
         return customMap;
     }
 
+    public static Map<String,String> getMapWithJoinedValueFromRequest(PortletRequest request, String key, String value) {
+        Map<String, Set<String>> customMap = getCustomMapFromRequest(request, key, value);
+        return customMap.entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey,
+                        e -> {
+                            try {
+                                if(!isNotEmpty(e.getValue()) && e.getValue().size() == 1) {
+                                    return e.getValue().stream()
+                                            .findFirst()
+                                            .orElse("");
+                                }
+                                ObjectMapper mapper = new ObjectMapper();
```

Comment:
Creating an object mapper is a rather expensive operation; so this should not be done for each element in the map. The object creation should at least be moved to the beginning of the method.

As the ObjectMapper class is thread-safe, in this case it would also be an option to have a static ObjectMapper field in PortletUtils that is always reused.

Reply:
Thanks a lot, @oheger-bosch 
I have refactored the code.

Reply:
@albert-hjy, thanks for the update. This looks better now performance-wise. I am, however, a bit worried that the lazy initialization approach is not thread-safe. As the utils class is called during request execution, there will be multiple threads involved, and we should take care that this is safe.

The easiest solution is certainly to get rid off the lazy initialization, make the ObjectMapper field *final*, and create the mapper directly.

If you prefer lazy initialization, then I would suggest using the [Lazy initialization holder class idiom](https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom) for static fields.

Please also refer to J. Bloch, "Effective Java", 3rd edition, item 83 "Use lazy initialization judiciously"

Reply:
@oheger-bosch , thanks a lot. Agree that it's unnecessary to use lazy initialization here. I have changed ObjectMapper field _final_.